metaflow/plugins/parallel_decorator.py

```diff
@@ -25,6 +47,75 @@ def step_init(
         self, flow, graph, step_name, decorators, environment, flow_datastore, logger
     ):
         self.environment = environment
+        # We choose `setattr` instead of the default `current._update_env` for two reasons: first, to preserve
```

Comment:
Could we then not do this in task_pre_step if that is when the values are available. Not strongly opposed to this but the _update_env basically does what this does so it feels like as long as we do it prior to the user code starting (so task_pre_step is early enough), we should be good too? Or am I missing another case.

Reply:
Well overall, I can keep the parallel property in the `current` object and nothing has to change here. 

This became a little essential because: 
- the `Parallel` namedtuple returned by `current.parallel` sets it's attributes using env vars. 
- The earlier implementation had the `Parallel` namedtuple returned via `current.parallel` property (the property was statically set). 
- Now we are moving to an implementation where `current.parallel` is dynamically set. 
- In this new world also want to retain the data structure of `Parallel` so that we have backwards compatibility in terms of types. 
- Now if we set this named tuple using `current._update_env` we will be _statically setting the namedtuple_. This doesn't work for us because the values inside the named tuple are derived using env vars. These env vars are set in `task_pre_step` based on the compute decorator. Since order of calling `task_pre_step` can vary, we need ensure that the namedtuple can have all the right data set to it and it is dynamically retrieving those values. Doing a `current._update_env` won't help us gaurentee that the values will always be correctly set in task_pre_step. 
- Hence setting this as an explicit property via setattr instead of passing it down via `current._update_env`